graphbolt/src/cnumpy.cc

```diff
@@ -88,145 +97,159 @@ void OnDiskNpyArray::ParseNumpyHeader(torch::Tensor shape) {
+                } else {
+                  read_size = aligned_length;
+                }
+                // Put requests into io_uring queue.
+                struct io_uring_sqe *submit_queue =
+                    io_uring_get_sqe(&io_uring_queue_[thread_id]);
+                io_uring_prep_read(
+                    submit_queue, file_description_,
+                    read_buffer +
+                        ((aligned_length + kDiskAlignmentSize) * group_size_ *
+                         thread_id) +
+                        ((aligned_length + kDiskAlignmentSize) *
+                         (group_id % group_size_)),
+                    read_size, aligned_offset);
+
+                if (((group_id + 1) % group_size_ == 0) || i == end - 1) {
+                  if (!error_flag.load()) {
+                    io_uring_submit(&io_uring_queue_[thread_id]);
+                    //  Wait for completion of I/O requests.
+                    int64_t num_finish = 0;
+                    // Wait until all the disk blocks are loaded in current
+                    // group.
+                    while (num_finish < (group_id % group_size_ + 1)) {
+                      struct io_uring_cqe *complete_queue;
+                      if (io_uring_wait_cqe(
+                              &io_uring_queue_[thread_id], &complete_queue) <
+                          0) {
+                        perror("io_uring_wait_cqe");
+                        std::exit(EXIT_FAILURE);
+                      }
+                      struct io_uring_cqe *complete_queues[group_size_];
+                      int cqe_count = io_uring_peek_batch_cqe(
+                          &io_uring_queue_[thread_id], complete_queues,
+                          group_size_);
+                      if (cqe_count == -1) {
+                        perror("io_uring_peek_batch error\n");
+                        std::exit(EXIT_FAILURE);
+                      }
+                      // Move the head pointer of completion queue.
+                      io_uring_cq_advance(
+                          &io_uring_queue_[thread_id], cqe_count);
+                      num_finish += cqe_count;
+                    }
+                    // copy results into result_buffer.
+
+                    for (int64_t batch_id = batch_offset; batch_id <= i;
+                         batch_id++) {
+                      std::memcpy(
+                          result_buffer + feature_size_ * (batch_id),
+                          read_buffer +
+                              ((aligned_length + kDiskAlignmentSize) *
+                               group_size_ * thread_id) +
+                              ((aligned_length + kDiskAlignmentSize) *
+                                   ((batch_id - begin) % group_size_) +
+                               residual
+                                   [thread_id * group_size_ +
+                                    ((batch_id - begin) % group_size_)]),
```

Comment:
name this 2 variables.

Reply:
Named some of these calculated values. Not sure if the naming is very accurate.